metaflow/includefile.py

```diff
@@ -301,81 +253,76 @@ class IncludeFile(Parameter):
     def __init__(
         self, name, required=False, is_text=True, encoding=None, help=None, **kwargs
     ):
-        # Defaults are DeployTimeField
+        # If a default is specified, it needs to be uploaded when the flow is deployed
+        # (for example when doing a `step-functions create`) so we make the default
+        # be a DeployTimeField. This means that it will be evaluated in two cases:
+        #  - by deploy_time_eval for `step-functions create` and related.
+        #  - by Click when evaluating the parameter.
+        #
+        # In the first case, we will need to fully upload the file whereas in the
+        # second case, we can just return the string as the FilePath.convert method
+        # will take care of evaluating things.
         v = kwargs.get("default")
         if v is not None:
-            _, file_type, _ = LocalFile.is_file_handled(v)
-            # Ignore error because we may never use the default
-            if file_type is None:
-                o = {"type": "self", "is_text": is_text, "encoding": encoding, "url": v}
-                kwargs["default"] = DeployTimeField(
-                    name,
-                    str,
-                    "default",
-                    lambda ctx, full_evaluation, o=o: LocalFile(
-                        o["is_text"], o["encoding"], o["url"]
-                    )(ctx)
-                    if full_evaluation
-                    else json.dumps(o),
-                    print_representation=v,
-                )
-            else:
-                kwargs["default"] = DeployTimeField(
-                    name,
-                    str,
-                    "default",
-                    lambda _, __, is_text=is_text, encoding=encoding, v=v: Uploader.encode_url(
-                        "external-default", v, is_text=is_text, encoding=encoding
-                    ),
-                    print_representation=v,
-                )
+            # The default may itself be a callable in which case we wrap it in
+            # a DeployTimeField just like Parameter does.
+            # In the end, we will have a DeployTimeField wrapping this DeployTimeField
+            # and we will properly extract in _eval_default.
```

Comment:
Will do, basically, the default for IncludeFile is *always* a DeployTimeField (since we need to do something at deploy time -- namely upload the file). If it is also a callable, that in itself is a DeployTimeField (since we need to call it at deploy time). When you have both, the inner DeployTimeField corresponds to calling the callable giving you some file name and the outer DeployTimeField is the upload portion.